CONTRIBUTING.md

```diff
@@ -185,3 +185,375 @@ work when changes to the proposed API become necessary.
 
 You can create a special issue or PR just for the sake of discussing a proposed
 API change. The actual implementation can follow later on.
+
+## Code Style
+
+###  General
+1. Avoid the use of [Magic Numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)#Unnamed_numerical_constants).
+   For example use `UA_STATUSCODE_BADINTERNALERROR` instead of `0x80020000`
+2. Avoid using global variables. If you for some reason really need to use global variables, prefix them with `g_` so that it is immediately clear in the code, that we are dealing with a global variable.
+3. Almost all of the code is indented by four (4) spaces and regardless of which is better, spaces or tabs, you should indent with four spaces in order to be consistent.
+4. If you find yourself copy-pasting code, consider refactoring it into a function and calling that.
+   Try keeping functions short (about 20-30 lines, that way they will in most cases fit onto the screen).
+   In case of simple switch statements for example this can be exceeded, but there is not much complexity that needs to be understood.
+   Most of the time, if functions are longer than 30 lines, they can be refactored into smaller functions that make their intent clear.
+5. Use of comments
+
+Use C-style comments as follows:
+
+```c
+/* Lorem ipsum dolor sit amet */
+
+/* Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy
+ * eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam
+ * voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita
+ * kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem
+ * ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod
+ * tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua.
+ * At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd
+ * gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. */
+```
+
+Note that public header files (in the /include directory) use comments beginning with a double-star to generate documentation. Just follow along the formatting that is already in place.
+
+```c
+/** This is documentation */
+```
+
+Every (good) editor can reflow comments for nice formatting.
+
+Do not use C++ comments (double-slash `//`).
```

Comment:
The file should follow its own advice. :-)
C++-style comments are used in some of the new text.

Reply:
fixed :)